cmake/setup.cmake

```diff
@@ -0,0 +1,70 @@
+##############################
+###     CONFIGURATIONS     ###
+##############################
+include(FeatureSummary)
+
+# cmake policies
+include(cmake/policies.cmake)
+
+# user functions
+include(cmake/functions.cmake)
+
+# flags and config options
+include(cmake/config/all.cmake)
+include(cmake/config/release.cmake)
+include(cmake/config/debug.cmake)
+include(cmake/config/coverage.cmake)
+
+# all 3rd-party dependencies
+# should be last in this section
+include(cmake/dependencies.cmake)
+
+
+
+##############################
+###        CCACHE          ###
```

Comment:
Why isn't it moved as well?

Reply:
moved where?

Reply:
Separate file, like analysis/*cmake

Reply:
ccache is not an analysis tool

Reply:
Like files in analysis, not into
I think `cmake/ccache.cmake` would be nice

Reply:
Okay, I will fix that.

Reply:
fixed